File: benchmarks/sieve.js
function sieve(n) {
    const flags = new Uint8Array(n + 1);
    let count = 0;
    for (let i = 2; i <= n; i++) {
        if (!flags[i]) {
            count++;
            for (let k = i * i; k <= n; k += i) {
                flags[k] = 1;
            }
        }
    }
    return count;
}

console.log(sieve(100000));

File: benchmarks/sieve.rs
fn sieve(n: usize) -> usize {
    let mut flags = vec![false; n + 1];
    let mut count = 0;
    for i in 2..=n {
        if !flags[i] {
            count += 1;
            let mut k = i * i;
            while k <= n {
                flags[k] = true;
                k += i;
            }
        }
    }
    count
}

fn main() {
    println!("{}", sieve(100_000));
}
